src/cmd.ts

```diff
@@ -29,39 +28,61 @@ import "./signalHandlers.js";
 import { doStartupValidation } from "./startup.js";
 import { parseTorrentFromFilename } from "./torrent.js";
 import { fallback } from "./utils.js";
+import { inspect } from "util";
 
-function processOptions(options): RuntimeConfig {
-	if (options.rssCadence) {
-		options.rssCadence = Math.max(ms(options.rssCadence), ms("10 minutes"));
-	}
-	if (options.searchCadence) {
-		options.searchCadence = ms(options.searchCadence);
+const fileConfig = await getFileConfig();
 
-		// Set a minimum cadence of once a day if excludeRecentSearch hasn't been set to avoid flooding requests
-		if (!options.excludeRecentSearch) {
-			options.searchCadence = Math.max(
-				options.searchCadence,
-				ms("1 day")
+/**
+ * validates and sets RuntimeConfig
+ * @return (the number of errors Zod encountered in the configuration)
+ */
+
+export async function validateAndSetRuntimeConfig(options: RuntimeConfig) {
+	initializeLogger(options);
+	logger.info(`${PROGRAM_NAME} v${PROGRAM_VERSION}`);
+	logger.info("Validating your configuration...");
+	try {
+		options = VALIDATION_SCHEMA.parse(options, {
+			errorMap: zodErrorMap,
+		}) as RuntimeConfig;
+	} catch (error) {
+		logger.verbose({
+			label: Label.CONFIGDUMP,
+			message: inspect(options),
+		});
+		error.errors.forEach(({ path, message }) => {
+			const urlPath = path.toString().toLowerCase().split(",")[0];
+			const optionLine = path.toString().includes(",")
+				? (() => {
+						const setting = path.toString().split(",");
+						const position = Number(setting[1].trim()) + 1;
+						return `${setting[0]} (position #${position})`;
+				  })()
+				: path;
+			logger.error(
+				`${
+					path.length > 0
```

Comment:
is path.length ever 0?

Reply:
When it's refined outside (for options like inject without a <client>Url etc - path is 0....basically anytime we validate without it being explicitly tied to a variable.

```
[ 'excludeOlder' ]
error:  Option:  excludeOlder
        Setting: "1dh"
        format does not follow vercel's ms style ( https://github.com/vercel/ms#examples )
        (https://www.cross-seed.org/docs/basics/options#excludeolder)

[]
error:  Configuration:
        You need to specify rtorrentRpcUrl, transmissionRpcUrl, qbittorrentUrl, or delugeRpcUrl when using 'inject'
        (https://www.cross-seed.org/docs/basics/options#)```

Reply:
Ah right, the top-level refines. Makes sense.